### PR TITLE
Add 'schedule status' column to admin indexer grid

### PR DESCRIPTION
**Pull request by fredden**
### Description (*)

This copies the 'schedule status' column from command line (`php bin/magento indexer:status`) into the admin grid for indexes.

### Manual testing scenarios (*)

1. Stop/disable Magento cron.
1. Set indexes to 'update on schedule'.
1. Note column in admin grid: system -> tools -> index management
1. Change product details which will need to be indexed.
1. Observe column in grid
1. Start/re-enable Magento cron.
1. Observe column in grid

Also compare admin grid content to command line content at relevant steps.

### Questions or comments

I do not know if this requires a unit test.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Indexer/Block/Backend/Grid/Column/Renderer/ScheduleStatus.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+use Magento\Framework\Phrase;
+
+class ScheduleStatus extends AbstractRenderer
+{
+    /**
+     * @var \Magento\Framework\Mview\ViewInterface
+     */
+    protected $viewModel;
+
+    public function __construct(
```
Comment (atwixfirster) [inline app/code/Magento/Indexer/Block/Backend/Grid/Column/Renderer/ScheduleStatus.php:38]: @fredden , can you please add the annotation block here? Thanks

Comment (fredden) [reply app/code/Magento/Indexer/Block/Backend/Grid/Column/Renderer/ScheduleStatus.php:38]: I've coped & modified the annotation block from the ancestor I'm overwriting, but the format doesn't match the other annotation block below. Please let me know if you want more/different content here.

Comment (atwixfirster) [reply app/code/Magento/Indexer/Block/Backend/Grid/Column/Renderer/ScheduleStatus.php:38]: Look ok for me. Thanks

**Comment by ihor-sviziev** [comment]
@magento run all tests

File: app/code/Magento/Indexer/Block/Backend/Grid/Column/Renderer/ScheduleStatus.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+/**
+ * Renderer for 'Schedule Status' column in indexer grid
+ */
+class ScheduleStatus extends AbstractRenderer
+{
+    /**
+     * @var \Magento\Framework\Escaper
+     */
+    protected $escaper;
```
Comment (ihor-sviziev) [inline app/code/Magento/Indexer/Block/Backend/Grid/Column/Renderer/ScheduleStatus.php]: Lease make all properties with private visibility

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @fredden,
Your changes mostly looks good to me.
Could you cover your changes with some tests? I think the best mftf test should be there

**Comment by engcom-Echo** [comment]
I will take care of test coverage.

**Review by ihor-sviziev — DISMISSED** [review]

**Comment by ihor-sviziev** [comment]
@engcom-Echo thank you!

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6977](https://jira.corp.magento.com/browse/ENGCOM-6977) has been created to process this Pull Request

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @fredden.

During testing, we faced the issue.

**_Problem:_** Schedule statuses for Design Config Grid and Customer Grid indexers are not displayed

**Manual testing scenario:**

1. Stop/disable Magento cron;
2. Set indexes to 'update on schedule';
3. Note column in admin grid: system -> tools -> index management

**Actual Result:** :heavy_multiplication_x:  Design Config Grid and Customer Grid indexers are empty
before
but if you check statuses in `php bin/magento indexer:status` , they were filled.
before1
2. Change product details that will need to be indexed.
3. Observe column in the grid

**Actual Result:** :heavy_multiplication_x:  The same situation with indexers
before2
b3

@fredden Could you take a look?

Thanks!

**Comment by ihor-sviziev** [comment]
Hi @fredden,
Will you be able to update your PR?
Your PR seems to me really good! I really don't want to close it due to inactivity

**Comment by fredden** [comment]
Yes, I will review and update.

**Review by ihor-sviziev — APPROVED** [review]
Approved.
Functional test failures are not related to changes from this PR

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6977](https://jira.corp.magento.com/browse/ENGCOM-6977) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Echo** [comment]
Functional test failures are not related to changes from this PR

**Pull request merged**